pallets/msa/src/lib.rs

```diff
@@ -905,6 +924,99 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Withdraw all available tokens from the account associated with the MSA, to the account of the caller.
+		///
+		/// The `origin` parameter represents the account from which the function is called and must be the account receiving the tokens.
+		///
+		/// The function requires one signature: `msa_owner_proof`, which serve as proof that the owner of the MSA authorizes the transaction.
+		///
+		/// The necessary information for the withdrawal authorization, the destination account and the MSA ID, are contained in the `authorization_payload` parameter of type [AddKeyData].
+		/// It also contains an expiration block number for the proof, ensuring it is valid and must be greater than the current block.
+		///
+		/// # Events
+		/// * [`pallet_balances::Event::<T,I>::Transfer`] - Transfer token event
+		///
+		/// # Errors
+		///
+		/// * [`Error::NotKeyOwner`] - the transaction signer origin is not the owner of the public key contained in the provided `authorization_payload`
+		/// * [`Error::MsaOwnershipInvalidSignature`] - `msa_owner_public_key` is not a valid signer of the provided `authorization_payload`.
+		/// * [`Error::NoKeyExists`] - the public key supplied in `msa_owner_public_key` is not a registered control key of any MSA.
+		/// * [`Error::NotMsaOwner`] - the MSA ID in `authorization_payload` does not match the MSA ID of the `msa_owner_public_key`.
+		/// * [`Error::ProofHasExpired`] - the current block is less than the `expired` block number set in `AddKeyData`.
+		/// * [`Error::ProofNotYetValid`] - the `expired` block number set in `AddKeyData` is greater than the current block number plus mortality_block_limit().
+		/// * [`Error::SignatureAlreadySubmitted`] - signature has already been used.
+		/// * [`Error::InsufficientBalanceToWithdraw`] - the MSA account has not balance to withdraw
+		/// * [`Error::InsufficientBalanceToFundDestination`] - the transfer would result in an account that would be reaped due to existential balance requirements
+		/// * [`Error::UnexpectedTokenTransferError`] - the balances pallet threw an error during the attempted transfer
+		///
+		#[pallet::call_index(14)]
+		#[pallet::weight(T::WeightInfo::withdraw_tokens())]
+		pub fn withdraw_tokens(
+			origin: OriginFor<T>,
+			msa_owner_public_key: T::AccountId,
+			msa_owner_proof: MultiSignature,
+			authorization_payload: AddKeyData<T>,
+		) -> DispatchResult {
+			let public_key = ensure_signed(origin)?;
```

Comment:
this must not be a control key since we don not allow control keys to have any tokens.